download-chromedriver.js

```diff
@@ -27,11 +27,25 @@ function processDownload (err, zipPath) {
   })
 }
 
-download(versionToDownload, (err, zipPath) => {
-  if (err) {
-    const parts = versionToDownload.split('.')
-    const baseVersion = `${parts[0]}.${parts[1]}.0`
-    download(baseVersion, processDownload)
+let primaryVersion = thisPackageVersion
+const parts = primaryVersion.split('.')
+let fallbackVersion = `${parts[0]}.${parts[1]}.0`
+
+const versionEnvVar = (
```

Comment:
I think it's pretty reasonable here to ask that developer follow a single convention: `ELECTRON_CHROMEDRIVER_CUSTOM_VERSION` will probably suffice. 

We can also probably DRY this code out a bit:

```js
const parts = thisPackageVersion.split('.')
const defaultVersion = `${parts[0]}.${parts[1]}.0`

const versionToDownload = process.env.ELECTRON_CHROMEDRIVER_CUSTOM_VERSION || defaultVersion

download(primaryVersion, (err, zipPath) => {
  if (err) {
    download(defaultVersion, processDownload)
  } else {
    processDownload(err, zipPath)
  }
...
```

Reply:
Thanks for the quick review!

I used the multiple conventions for consistency with how `electron-download` handles the `ELECTRON_MIRROR`/`ELECTRON_CUSTOM_DIR`/`ELECTRON_CUSTOM_FILENAME` variables already (see [electron/get's artifact-utils.ts](https://github.com/electron/get/blob/master/src/artifact-utils.ts#L27)); my understanding is that the `npm_*` variants, in particular, are there to allow for the use of a `.npmrc` file rather than an environment variable, to aid teams that want to share the use of a specific mirror across all their devs/environments. https://github.com/electron/get/pull/21 suggests that the weird casing variants exist as a workaround for some "exciting" npm behavior with these config files/variables.

The environment variable is the only thing my specific team needs, though, so if you value the simplicity more than the consistency, I'm happy to make the switch.

---

Avoiding the fallback version in the env-var case was an intentional decision; I think the fallback behavior makes sense in the context of the default behavior with the library version, but I expect that users that would want to use this override would mostly be doing so for either

* compliance reasons (where strictly constraining the version is usually a requirement), or
* compatibility reasons (where unexpectedly using mismatched chromedriver/chromium versions can cause difficult-to-diagnose test flakiness issues)

...and I think in either of those cases it would be more user-friendly to fail fast rather than falling back if the specifically requested version is not available.

Again, I'm happy to make the switch if you feel strongly about it in the other direction; I'm sure there are user scenarios I haven't understood here. Do you have a particular use case in mind that would benefit from the combination of an explicit version override with the fallback behavior?

Reply:
@codebytere Friendly ping - have you had a chance to decide which direction you'd like to go here? I'm happy to go with whichever direction you think is more appropriate for the project, I just wanted to explain why I had made the original decisions before making the changes.

Thanks!